http/client.go

```diff
@@ -144,53 +152,47 @@ func New(options ...Option) *Client {
 	}
 
 	return &Client{
-		Client:          *http.DefaultClient,
+		Client:          *opts.httpClient,
 		headersFunc:     opts.headersFunc,
 		maxSendAttempts: MaxSendAttempts,
 	}
 }
 
 // NewNonSSLValidating creates a new goose http *Client skipping SSL validation.
+// DEPRECATED (stickupkid): No longer used internally, left to ensure we don't
+// break compatibility.
```

Comment:
Can we move to a v3, break compatibility and clean more up?  Especially with a few client methods needing to return an error.